wasm/tx_become_validator/src/lib.rs

```diff
@@ -6,12 +6,19 @@ use namada_tx_prelude::transaction::pos::BecomeValidator;
 use namada_tx_prelude::*;
 
 #[transaction]
-fn apply_tx(ctx: &mut Ctx, tx_data: Tx) -> TxResult {
-    let signed = tx_data;
-    let data = signed.data().ok_or_err_msg("Missing data").map_err(|err| {
-        ctx.set_commitment_sentinel();
-        err
-    })?;
+fn apply_tx(ctx: &mut Ctx, tx_data: BatchedTx) -> TxResult {
+    let BatchedTx {
+        tx: signed,
+        ref cmt,
+    } = tx_data;
+    let data =
+        signed
+            .data(cmt)
+            .ok_or_err_msg("Missing data")
+            .map_err(|err| {
+                ctx.set_commitment_sentinel();
+                err
+            })?;
```

Comment:
we should abstract this a bit as it's repeated in every tx, maybe something like:

```rust
let data = ctx.get_tx_data(tx_data)?;
```

in tx_prelude e.g.:
```rust
impl Ctx {
  pub fn get_tx_data(&mut self, tx_data: BatchedTx) -> TxResult {
    ...
  }
}
```

Reply:
Agree, this would be a nice improvement!

Reply:
Done in 0a40071e8fddbf813703f9131af0772ac4a98e91

Reply:
nice!